pkg/controller/servicebindingrequest/binding_test.go

```diff
@@ -330,7 +367,7 @@ func TestServiceBinder_Bind(t *testing.T) {
 				},
 				ResourceRef: d.GetName(),
 			},
-			BackingServiceSelectors: []v1alpha1.BackingServiceSelector{
+			BackingServiceSelectors: &[]v1alpha1.BackingServiceSelector{
```

Comment:
The reason I prefer using pointers is that it helps our code distinguish between the following:
- if the user has specified an empty field 
- if the user has skipped the field

When `*BackingServiceSelectors` is used:
- when nil, we know that the field was skipped in the CR.
- when `[]` we know that an empty list was passed in the CR.

When `BackingServiceSelectors` is used:
- when [], the field might have been skipped in the CR. OR the user might have specified an empty list in the CR <-- this is an ambiguity.

To avoid this ambiguity, I prefer using pointers for optional fields because it lets us write code to handle scenarios as applicable.

Moreover,  https://github.com/redhat-developer/service-binding-operator/pull/369